ci-operator/step-registry/gather/proxy/gather-proxy-commands.sh

```diff
@@ -0,0 +1,54 @@
+#!/bin/bash
+set -o nounset
+set -o errexit
+set -o pipefail
+
+trap 'CHILDREN=$(jobs -p); if test -n "${CHILDREN}"; then kill ${CHILDREN} && wait; fi' TERM
+
+export AWS_SHARED_CREDENTIALS_FILE="${CLUSTER_PROFILE_DIR}/.awscred"
+
+if test ! -f "${SHARED_DIR}/proxyregion"
+then
+	echo "No proxyregion, so unknown AWS region, so unable to tear down ."
+	exit 0
+fi
+
+REGION="$(cat "${SHARED_DIR}/proxyregion")"
+PROXY_NAME="${NAMESPACE}-${JOB_NAME_HASH}"
+
+# cleaning up after ourselves
+aws --region "${REGION}" s3 rb "s3://${PROXY_NAME}" --force
+
+STACK_NAME="${PROXY_NAME}-proxy"
+
+if test -f "${SHARED_DIR}/proxy-instance-ids.txt"
+then
+  cat "${SHARED_DIR}/proxy-instance-ids.txt" | sort | uniq | while read -r INSTANCE_ID
+  do
+	  echo "Gathering console logs for ${INSTANCE_ID}"
+	  aws --region "${REGION}" ec2 get-console-output --instance-id "${INSTANCE_ID}" --output text > "${ARTIFACT_DIR}/${INSTANCE_ID}" &
+	  wait "$!"
+  done
+fi
+
+# collect logs from the proxy here
+if [ -f "${SHARED_DIR}/proxyip" ]; then
+  proxy_ip="$(cat ${SHARED_DIR}/proxyip)"
+  proxy_dir="${ARTIFACT_DIR}/proxy"
```

Comment:
No need for this subdir.  `${ARTIFACT_DIR}` is already unique per-step (e.g. [here][1]), so you can write straight to `${ARTIFACT_DIR}/squid.service`.

[1]: https://storage.googleapis.com/origin-ci-test/pr-logs/pull/openshift_release/9904/rehearse-9904-pull-ci-openshift-installer-master-e2e-aws-proxy/1280141908869386240/artifacts/e2e-aws-proxy/gather-proxy/proxy/squid.service